src/app/components/Application/Application.jsx

```diff
@@ -15,18 +15,29 @@ import {
 
 import { breakpoints } from '../../data/constants';
 import DataLoader from '../DataLoader/DataLoader';
+import appConfig from '../../data/appConfig';
 
 class Application extends React.Component {
-  constructor(props) {
-    super(props);
+  constructor(props, context) {
+    super(props, context);
+    const {
+      query,
+    } = context.router.location;
     this.state = {
       data: Store.getState(),
       patron: PatronStore.getState(),
       media: 'desktop',
     };
     this.onChange = this.onChange.bind(this);
-    this.shouldStoreUpdate = this.shouldStoreUpdate.bind(this);
     this.submitFeedback = this.submitFeedback.bind(this);
+
+    const urlEnabledFeatures = query.features ? query.features.split(',') : null;
+    if (urlEnabledFeatures) {
+      const urlFeaturesString = urlEnabledFeatures.filter(
+        urlFeat => !appConfig.features.includes(urlFeat))
```

Comment:
Yes

Reply:
Is there a reason to distinguish url-enabled features from those enabled through other means? Within a component, do we care whether the feature is enabled by URL or via appConfig? Wondering if Application should maintain a state var like `state.features`, which is the union of features enabled by URL / appConfig. Unless you think there's a reason to distinguish..

Reply:
I distinguished between them, because when the feature is enabled by environment variable, there is no secret redirect. So, this is more efficient if someone happened to use the URL variable for a feature already enabled in the config.

Reply:
I think this may be one reason to prefer setting a cookie. Having a lot of redirects works, but seems inefficient.

Reply:
Yeah, I thought it might too, but it appears seamless locally, even with login. A bigger overhaul would be creating a custom link component or push function to use everywhere in the app and give us a place to append this to all links routed to in the application.